### PR TITLE
[APP-6886] Add option to reboot device after a certain amount of time

### PR DESCRIPTION
**Pull request by ale7714**

### CODE REVIEW

File: subsystems/provisioning/definitions.go
Comment (ale7714) [inline subsystems/provisioning/definitions.go:382]: I consider doing `device_reboot_after_offline_hours` while not super granular. I also can't imagine you would want less than one hour

Comment (Otterverse) [inline subsystems/provisioning/definitions.go:382]: Would prefer to see this be a `Timeout` instead of `int` as then the json gets parsed automatically for timeouts.

File: subsystems/provisioning/definitions.go
Comment (Otterverse) [inline subsystems/provisioning/definitions.go:304]: Would prefer to see native time parsing, not converting stuff to ints. Look at the other timeout(s) above. If the type here is Timeout, all the parsing happens automagically, so no need to cast and stuff here. It can just follow the pattern of the other timeouts (directly above.)

File: subsystems/provisioning/definitions.go
Comment (Otterverse) [reply subsystems/provisioning/definitions.go:382]: We may need to change the default parsing to "minutes" for the timeout values (currently it assumes bare numbers are seconds.)

File: subsystems/provisioning/networkmanager.go
Comment (Otterverse) [inline subsystems/provisioning/networkmanager.go:700]: This feels very dangerous. This feels like it'd be a rugpull to running processes. Equivalent to `SYSREQ+b` Does it actually do a normal shutdown? If not, maybe exec "systemctl reboot" to guarantee a normal restart.

File: subsystems/provisioning/definitions.go
Comment (Otterverse) [reply subsystems/provisioning/definitions.go:382]: Minutes makes the most sense. All the timeouts should have the same "unit" and minutes feels correct. Also, with the way Timeout parsing works, you can use decimal units if needed. Or "1h"... take a look at the Timeout.Unmarshall() function. It treats bare numbers as seconds currently, but we can change that to minutes, but people can specify easier units if they want. "10h30m" and such (we don't have to document/advertise that extra feature!)

**Review by Otterverse — CHANGES_REQUESTED** [review]
A few notes before I hit the road... No need to wait for me if you need to merge this before I get back on Tuesday though.

File: subsystems/provisioning/networkmanager.go
Comment (ale7714) [reply subsystems/provisioning/networkmanager.go:700]: I believe is the same according to the docs, underhood systemctl is running reboot but will update to use systemctl for consistency 

https://www.man7.org/linux/man-pages/man1/systemctl.1.html

**Review by Otterverse — CHANGES_REQUESTED** [review]
More notes... sorry I didn't get all of these noted last time.

**Comment by ale7714** [comment]
sorry removing review request because i kind of push some changes quickly but did not test or anything, and then got sidetracked

**Comment by ale7714** [comment]
ready for review now

File: subsystems/provisioning/networkmanager.go
Comment (Otterverse) [inline subsystems/provisioning/networkmanager.go:753]: Can't do this, or agent itself won't shut down when called and will just hang until the sleep is done. It will also fail healthchecks if the loop pauses here for too long. There's a function for pausing the loop though. See line 623 above for the previous use of the following:

```
if !w.mainLoopHealth.Sleep(ctx, time.Second*100) {
	return
}
w.logger.Errorf("Failed to reboot after %s time!", time.Second * 100)			
```

May also need a slightly longer timeout, as I BELIEVE that systemd 90 second default is per service, so if it's shutting down multiple services, it can definitely take longer than that. I'd think 2 or maybe even 5 minutes, since this isn't going to handle the failure case (where reboot doesn't happen.) Regardless, I'd DEFINITELY suggest logging an error if the timeout expires (included in suggestion above... rewrite as needed) so the user knows things are off the rails.

**Review by Otterverse — CHANGES_REQUESTED** [review]
Getting there... couple more gotchas though.

File: subsystems/provisioning/networkmanager.go
Comment (ale7714) [reply subsystems/provisioning/networkmanager.go:753]: makes sense, thanks i missed there's a sleep on the main loop

**Review by Otterverse — APPROVED** [review]
Realized this was fixed, but didn't see a review request. LGTM anyway, but if you have more changes planned, go ahead.

**Pull request merged**
